pkg/kubectl/cmd/swagger_doc.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+// NewCmdSwaggerDoc returns a cobra command for swagger docs
+func NewCmdSwaggerDoc(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "doc model MODEL-NAME | operation URL [GET|POST|PUT|DELETE|PATCH]",
+		Short: "Documentation of models and operations. API version in models can be defined as a prefix: v1.MODEL. For operations is embedded in the URL.",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunSwaggerDoc(f, out, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmd.Flags().BoolP("search", "S", false, "Search for similar models to given MODEL-NAME or URL")
+	return cmd
+}
+
+// RunSwaggerDoc executes the appropriate steps to print a model's documentation
+func RunSwaggerDoc(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	if !(args[0] == "model" || args[0] == "operation") || len(args) > 3 || len(args) == 0 {
+		return cmdutil.UsageError(cmd, "We accept only this format: doc model MODEL-NAME | operation URL [GET|POST|PUT|DELETE|PATCH]")
+	}
+
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+
+	mode := args[0]
+	search := cmdutil.GetFlagBool(cmd, "search")
+	apiV := kubectl.GetAPIVersion(args[1:], mode)
```

Comment:
There is an `--api-version` flag you inherit from kubectl, use that to get the api version